Getting async leaking error when testing opine v2

# Issue: Getting async leaking error when testing opine v2

Setup:

- Deno Version: 1.16.4
- v8 Version: 9.7.16.15
- Typescript Version: 4.4.2
- SuperDeno Version: 4.7.1

## Details

> When running a simple test with opine@2.0.0 Deno Test Assertions return `AssertionError: Test case is leaking async ops.` - there are no async's in the example code which is pasted below.

server.ts

``` js
import { opine } from 'https://deno.land/x/opine@2.0.0/mod.ts'


export function build() {

 const app = opine();

  app.get('/', function (req, res) {
    res.send('Hello World')
  })

  return app
}
```

test.ts

``` js
import {superdeno} from 'https://deno.land/x/superdeno@4.7.1/mod.ts'
import { build } from './server.ts'
const app = build()

Deno.test('hello world', () => {
  superdeno(app)
    .get('/')
    .expect(200)
})
```

Run

``` sh
deno test -A --unstable test.ts
```

Output

``` sh
running 1 test from file:///workspace/deno-playground/test.ts
test hello world ... FAILED (12ms)

failures:

hello world
AssertionError: Test case is leaking async ops.
Before:
  - dispatched: 0
  - completed: 0
After:
  - dispatched: 2
  - completed: 1
Ops:
  op_net_accept:
    Before:
      - dispatched: 0
      - completed: 0
    After:
      - dispatched: 1
      - completed: 0

Make sure to await all promises returned from Deno APIs before
finishing test case.
    at assert (deno:runtime/js/06_util.js:41:13)
    at asyncOpSanitizer (deno:runtime/js/40_testing.js:128:7)
    at async resourceSanitizer (deno:runtime/js/40_testing.js:144:7)
    at async Object.exitSanitizer [as fn] (deno:runtime/js/40_testing.js:176:9)
    at async runTest (deno:runtime/js/40_testing.js:381:7)
    at async Object.runTests (deno:runtime/js/40_testing.js:494:22)

failures:

        hello world

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out (147ms)

error: Test failed

```

--- a/src/test.ts
+++ b/src/test.ts
@@ -472,6 +472,10 @@ export class Test extends SuperRequest {
 
         return close(self.#server, self.app, undefined, async () => {
           await completeXhrPromises();
+
+          // REF: https://github.com/denoland/deno/blob/987716798fb3bddc9abc7e12c25a043447be5280/ext/timers/01_timers.js#L353
+          await new Promise((resolve) => setTimeout(resolve, 20));
+
           self.#assert(err, res, callback);
         });
       },

--- a/test/superdeno.opine.test.ts
+++ b/test/superdeno.opine.test.ts
@@ -321,12 +321,24 @@ describe("superdeno(app)", () => {
         res.send("superdeno FTW!");
       });
 
+      let doneCount = 0;
+
       superdeno(app)
         .get("/")
-        .end(() => {});
+        .end(() => {
+          doneCount++;
+
+          if (doneCount === 2) {
+            done();
+          }
+        });
 
       app.on("close", () => {
-        done();
+        doneCount++;
+
+        if (doneCount === 2) {
+          done();
+        }
       });
     });
 

